nuxeo-features/nuxeo-annotation/nuxeo-annotation-core/src/main/java/org/nuxeo/ecm/annotation/AnnotationServiceImpl.java

```diff
@@ -0,0 +1,128 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Funsho David
+ *
+ */
+
+package org.nuxeo.ecm.annotation;
+
+import static org.nuxeo.ecm.annotation.AnnotationConstants.ANNOTATION_COLOR_PROPERTY;
+import static org.nuxeo.ecm.annotation.AnnotationConstants.ANNOTATION_CREATION_DATE_PROPERTY;
+import static org.nuxeo.ecm.annotation.AnnotationConstants.ANNOTATION_DATE_PROPERTY;
+import static org.nuxeo.ecm.annotation.AnnotationConstants.ANNOTATION_DOCUMENT_ID_PROPERTY;
+import static org.nuxeo.ecm.annotation.AnnotationConstants.ANNOTATION_DOC_TYPE;
+import static org.nuxeo.ecm.annotation.AnnotationConstants.ANNOTATION_FLAGS_PROPERTY;
+import static org.nuxeo.ecm.annotation.AnnotationConstants.ANNOTATION_ID_PROPERTY;
+import static org.nuxeo.ecm.annotation.AnnotationConstants.ANNOTATION_LAST_MODIFIER_PROPERTY;
+import static org.nuxeo.ecm.annotation.AnnotationConstants.ANNOTATION_NAME_PROPERTY;
+import static org.nuxeo.ecm.annotation.AnnotationConstants.ANNOTATION_OPACITY_PROPERTY;
+import static org.nuxeo.ecm.annotation.AnnotationConstants.ANNOTATION_PAGE_PROPERTY;
+import static org.nuxeo.ecm.annotation.AnnotationConstants.ANNOTATION_POSITION_PROPERTY;
+import static org.nuxeo.ecm.annotation.AnnotationConstants.ANNOTATION_SECURITY_PROPERTY;
+import static org.nuxeo.ecm.annotation.AnnotationConstants.ANNOTATION_SUBJECT_PROPERTY;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.nuxeo.ecm.core.api.CoreSession;
+import org.nuxeo.ecm.core.api.DocumentModel;
+import org.nuxeo.ecm.core.api.DocumentModelList;
+import org.nuxeo.ecm.core.api.IdRef;
+import org.nuxeo.runtime.model.DefaultComponent;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * @since 10.1
+ */
+public class AnnotationServiceImpl extends DefaultComponent implements AnnotationService {
+
+    private static final Log log = LogFactory.getLog(AnnotationServiceImpl.class);
+
+    protected static final String ANNOTATIONS_FROM_DOCUMENT = "SELECT * FROM Annotation WHERE annotation:annotationId LIKE '%s%%' ";
+
+    @Override
+    public String createAnnotation(DocumentModel documentModel, Annotation annotation) {
+        CoreSession session = documentModel.getCoreSession();
+        String id = documentModel.getRepositoryName() + "-" + documentModel.getId() + "-" + annotation.getDocumentId();
+
+        // Create annotation as a placeless document
+        DocumentModel annotationModel = session.createDocumentModel(null, id, ANNOTATION_DOC_TYPE);
+        annotationModel.setPropertyValue(ANNOTATION_ID_PROPERTY, id);
+        setAnnotationProperties(annotationModel, annotation);
+        annotationModel = session.createDocument(annotationModel);
+        session.save();
```

Comment:
We have to decide if it's really a good idea to do `session.save()` here, as it's a costly operation. Given that we manipulate a `DocumentModel` in this API I think it's clear to the caller (and it can be explained in the Javadoc) that it's up to the caller to do a `session.save()` when required.